godot-core/src/builtin/file.rs

```diff
@@ -0,0 +1,459 @@
+    }
+
+    pub fn store_float(&mut self, value: f32) -> IoResult<()> {
+        self.fa.store_float(value);
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_real(&mut self, value: f32) -> IoResult<()> {
+        self.fa.store_real(value);
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_string(&mut self, value: GodotString) -> IoResult<()> {
+        self.fa.store_string(value);
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_pascal_string(&mut self, value: GodotString) -> IoResult<()> {
+        self.fa.store_pascal_string(value);
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_line(&mut self, value: GodotString) -> IoResult<()> {
+        self.fa.store_line(value);
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_csv_line(
+        &mut self,
+        values: PackedStringArray,
+        delim: GodotString,
+    ) -> IoResult<()> {
+        self.fa.store_csv_line_ex(values).delim(delim).done();
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_variant(&mut self, value: Variant, full_objects: bool) -> IoResult<()> {
+        self.fa
+            .store_var_ex(value)
+            .full_objects(full_objects)
+            .done();
+        self.check_error()?;
+        Ok(())
+    }
+
+    // Varia
+    pub fn set_big_endian(&mut self, value: bool) {
+        self.fa.set_big_endian(value);
+    }
+
+    pub fn is_big_endian(&self) -> bool {
+        self.fa.is_big_endian()
+    }
+
+    pub fn get_path(&self) -> GodotString {
+        self.fa.get_path()
+    }
+
+    pub fn get_path_absolute(&self) -> GodotString {
+        self.fa.get_path_absolute()
+    }
+
+    /// Returns the current file position
+    pub fn get_position(&self) -> u64 {
+        self.fa.get_position()
+    }
+}
+
+impl Read for GFile {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        let length = self.fa.get_length();
+        let position = self.fa.get_position();
+
+        if position >= length {
+            return Ok(0);
+        }
+
+        let remaining_bytes = (length - position) as usize;
+        let bytes_to_read = cmp::min(buf.len(), remaining_bytes);
+
+        if bytes_to_read == 0 {
+            return Ok(0);
+        }
+        let mut readen = 0;
+        while readen < bytes_to_read {
+            buf[readen] = self.fa.get_8();
+            readen += 1;
+            self.check_error()?;
+        }
+        Ok(readen)
+    }
+}
+
+impl Write for GFile {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let bytes_to_write = buf.len();
+        let mut bytes_written = 0;
+
+        while bytes_written < bytes_to_write {
+            self.fa.store_8(buf[bytes_written]);
+            bytes_written += 1;
+            self.check_error()?;
+        }
+        Ok(bytes_written)
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
+    }
```

Comment:
This call need not be forwarded?

Reply:
`flush()` in `Write` is tricky. There was a discussion in rust repo about making the default implementation as above. Not every writer needs one, this implementation surely doesn't, as it doesn't have internal buffer and just writes  byte by byte.

Reply:
Could you add a TODO so we don't forget it when optimizing multi-byte reads/writes?